lib/chef/provisioning/aws_driver/aws_provider.rb

```diff
@@ -22,8 +23,184 @@ def region
+            Chef::Log.debug "#{new_resource.to_s} is already associated with #{entry_id} in #{entry_driver.driver_url}"
+          else
+            Chef::Log.warn "#{new_resource.to_s} is currently associated with #{entry_id} in #{entry_driver.driver_url}, but the desired ID is #{desired_id} in #{new_resource.driver.driver_url}!  Will associate with new desired ID #{desired_id}."
+          end
+
+        else
+
+          #
+          # If we don't have desired (common case), we'll update the existing
+          # resource or create a new one if it's been deleted.
+          #
+          aws_object = new_resource.class.get_aws_object(entry_id, driver: entry_driver, resource: new_resource, required: false)
+          if aws_object
+            Chef::Log.debug "#{new_resource.to_s} is currently associated with #{entry_id} in #{entry_driver.driver_url}."
+          else
+            Chef::Log.warn "#{new_resource.to_s} is currently associated with #{entry_id} in #{entry_driver.driver_url}, but it does not exist!  We will create a new one to replace it."
+          end
+        end
+
+      else
+
+        #
+        # If we don't currently have an AWS ID associated with this resource, we
+        # will either associate the desired one, or create a new one.
+        #
+        if desired_id
+          Chef::Log.debug "#{new_resource.to_s} is not yet associated with anything.  Associating with desired object #{desired_id} in #{desired_driver.driver_url}."
+        else
+          Chef::Log.debug "#{new_resource.to_s} is not yet associated with anything.  Creating a new one in #{desired_driver.driver_url} ..."
+        end
+      end
+
+    else
+
+      #
+      # If it does not support storing IDs in Chef at all, just grab the existing
+      # object and we'll update (or not) based on that.
+      #
+      aws_object ||= new_resource.aws_object
+
+    end
+
+    #
+    # Actually update or create the AWS object
+    #
+    if aws_object
+      action, new_obj = update_aws_object(aws_object)
+      if action == :replaced_aws_object
+        aws_object = new_obj
+      end
+    else
+      aws_object = create_aws_object
+    end
+
+    #
+    # Associate the managed entry with the AWS object
+    #
+    if new_resource.is_a?(AWSResourceWithEntry)
+      new_resource.save_managed_entry(aws_object, action_handler, existing_entry: entry)
+    end
+
+    aws_object
+  end
+
+  action :destroy do
+    desired_driver = new_resource.driver
+    desired_id = new_resource.public_send(new_resource.class.aws_id_attribute) if new_resource.class.aws_id_attribute
+
+    #
+    # If the user specified an ID, delete THAT; do NOT delete the associated object.
+    #
+    if desired_id
+      aws_object = new_resource.class.get_aws_object(desired_id, resource: new_resource, required: false)
+      if aws_object
+        Chef::Log.debug "#{new_resource.to_s} provided #{new_resource.class.aws_id_attribute} #{desired_id} in #{desired_driver.driver_url}.  Will delete."
```

Comment:
I think you want `(#{desired_id})` to be consistent with your old log messages (id in parens)